ballerina/annotation_processor.bal

```diff
@@ -62,7 +62,7 @@ isolated function getSchemaString(GraphqlServiceConfig? serviceConfig) returns s
 }
 
 isolated function getServiceInterceptors(GraphqlServiceConfig? serviceConfig)
-    returns readonly & (readonly & Interceptor)[] {
+    returns readonly & ((readonly & Interceptor)|(readonly & Interceptor)[]) {
     if serviceConfig is GraphqlServiceConfig {
         return serviceConfig.interceptors;
     }
```

Comment:
To simplify the code, we can always return an interceptor array here.

```suggestion
    returns (readonly & Interceptor)[] {
    if serviceConfig is GraphqlServiceConfig {
        (readonly & Interceptor)|(readonly & Interceptor)[] interceptors = serviceConfig.interceptors;
        if serviceConfig.interceptors is (readonly & Interceptor) {
            return [interceptors];
        }
        return interceptors;
    }
```

Reply:
Fixed via https://github.com/ballerina-platform/module-ballerina-graphql/pull/1245/commits/0e79ac5618bfd988b2175f4d6fe6e96da485e4af